app/controllers/topics_controller.rb

```diff
@@ -848,7 +848,23 @@ def bulk
   end
 
   def reset_new
-    current_user.user_stat.update_column(:new_since, Time.now)
+    if params[:category_id].present?
+      category_ids = [params[:category_id]]
+
+      if params[:include_subcategories] == 'true'
+        category_ids = category_ids.concat(Category.where(parent_category_id: params[:category_id]).pluck(:id))
+      end
+
+      category_ids.each do |category_id|
```

Comment:
Is there a way we could do this in 1 SQL query instead of 1 per category?

Reply:
I think the first_or_initialize makes this somewhat tricky. I guess we could work this in 2 batches (existing category users followed by new category user records) however I feel it does make the code a fair bit more complex to reason about.

This should somewhat work even if a category has say 50-100 sub-categories, would start being really really painful if you try dismissing and there 10,000 sub categories. 

I think we can wait and see a bit here.